Automattic/jetpack
# Modify masterbar module to include new unified admin color schemes file

Alternative to https://github.com/Automattic/jetpack/pull/17750/.

As discussed in p1604929101083300-slack-ajax this PR repurposes and extends the existing Masterbar module to become a more all purpose module for any customizations required to the WPAdmin Dashboard experience on WP.com.

Specifically, where there was previously only the `A8C_WPCOM_Masterbar` class, we now add an additional `A8C_WPCOM_Unified_Admin_Color_Schemes` class and init that from the main file as well.

In the future, other classes could be added.

Please note that @obenland has already started work on customising the masterbar module here so we could look to land that. However, as it stands my PR (ie: this one) contains less code and is more "bare bones" and thus might be easier to merge.

#### Changes proposed in this Pull Request:

* Add `A8C_WPCOM_Unified_Admin_Color_Schemes` class to masterbar module and init from main file.
* Repurpose master bar module to become an all-purpose module for any customizations required to the WPAdmin Dashboard experience on WP.com.
* Add PHP `namespace` to the module.
* Move individual concerns into separate directories (eg: `/masterbar`, `/unified-admin-color-schemes`...etc).

#### Jetpack product discussion

* paYJgx-17t-p2#comment-1391
* p1604929101083300-slack-ajax

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

* Checkout PR
* Modify `init` method of `modules/masterbar/unified-admin-color-schemes.php` to include some debug output.
* Check that debug output is shown _only_ on admin pages.
* Verify existing masterbar works as previously.

#### Proposed changelog entry for your changes:
* Repurpose master bar module to become an all-purpose module for any customizations required to the WPAdmin Dashboard experience on WP.com.
* Add unified color schemes `class` to masterbar module and init from main file.
* Add PHP `namespace` to the module.
* Move individual concerns into separate directories (eg: `/masterbar`, `/unified-admin-color-schemes`...etc).
